src/main/webapp/app/course/manage/course-management-exercises.component.ts

```diff
@@ -28,14 +36,47 @@ export class CourseManagementExercisesComponent implements OnInit {
     ngOnInit(): void {
         this.courseId = Number(this.route.parent!.snapshot.paramMap.get('courseId'));
         this.courseService.find(this.courseId).subscribe((courseResponse) => (this.course = courseResponse.body!));
+        this.exerciseFilter = new ExerciseFilter('');
     }
 
     /**
-     * Sets the programming exercise count. Required to pass a callback to the
+     * Sets the (filtered) programming exercise count. Required to pass a callback to the
      * overrideProgrammingExerciseCard extension since extensions don't support @Output
      * @param count to set the programmingExerciseCount to
      */
     setProgrammingExerciseCount(count: number) {
         this.programmingExercisesCount = count;
     }
+    setFilteredProgrammingExerciseCount(count: number) {
+        this.filteredProgrammingExercisesCount = count;
+    }
+
+    /**
+     * Toggles the search bar
+     */
+    toggleSearch() {
+        this.showSearch = !this.showSearch;
+        // Reset the filter when the search bar is closed
```

Comment:
Why is this even necessary? I rather think that we shouldn't reset collapsables. This doesn't seem intuitive to me.

Reply:
Since I will already move the Search button to the bottom I can replace this behaviour with a "Reset" button

Reply:
Nice idea 👍